haskell/haddock.bzl

```diff
@@ -113,7 +113,7 @@ def _haskell_doc_aspect_impl(target, ctx):
   module_htmls = [
     ctx.actions.declare_file(
       paths.replace_extension(
-        path_to_module(ctx, f, sep='-', prefix=ctx.rule.attr.src_strip_prefix),
```

Comment:
Is `ctx.rule.attr.src_strip_prefix` different from `ctx.attr.src_strip_prefix` here? If so, we need to test this functionality, so the tests would fail on this change.

Reply:
`ctx.rule.attr` accesses attributes of a target we're applying [aspects](https://docs.bazel.build/versions/master/skylark/aspects.html) to. If you're in an aspect context, `ctx.attr` will refer to the attribute of the aspect itself rather than the target.

This makes a bunch of code very awkward sadly so you get keyword args like the one you see above. However now thinking about it maybe we can get rid of that and pass `ctx.rule` instead of `ctx` and it will just work. I would encourage you to try that so we can drop this `None` business.